benches/plonk.rs

```diff
@@ -1,147 +1,139 @@
-        let (pk_p, verifier_data) =
-            circuit.compile(&pp).expect("Compilation err");
-        let description = format!("Prove 2^{} = {} gates", degree, 1 << degree);
+    .expect("Unable to sample public parameters.");
 
-        c.bench_with_input(
-            BenchmarkId::new(description.as_str(), degree),
+    let mut compiling_benchmarks = c.benchmark_group("compile");
+    for degree in MINIMUM_DEGREE..MAXIMUM_DEGREE {
+        let mut circuit = BenchCircuit::<_, EdwardsParameters>::new(degree);
+        compiling_benchmarks.bench_with_input(
+            BenchmarkId::from_parameter(degree),
             &degree,
-            |b, _degree| {
+            |b, _| {
                 b.iter(|| {
-                    circuit
-                        .gen_proof(
-                            black_box(&pp),
-                            black_box(pk_p.clone()),
-                            black_box(&label),
-                        )
-                        .unwrap()
+                    circuit.compile(&pp).expect("Unable to compile circuit.")
                 })
             },
         );
+    }
+    compiling_benchmarks.finish();
 
-        let description =
-            format!("Verify 2^{} = {} gates", degree, 1 << degree);
-        let proof = circuit
-            .gen_proof(
-                black_box(&pp),
-                black_box(pk_p.clone()),
-                black_box(&label),
-            )
-            .unwrap();
+    let mut proving_benchmarks = c.benchmark_group("prove");
+    for degree in MINIMUM_DEGREE..MAXIMUM_DEGREE {
+        let mut circuit = BenchCircuit::<_, EdwardsParameters>::new(degree);
+        let (pk_p, _) =
+            circuit.compile(&pp).expect("Unable to compile circuit.");
+        proving_benchmarks.bench_with_input(
+            BenchmarkId::from_parameter(degree),
+            &degree,
+            |b, _| {
+                b.iter(|| circuit.gen_proof(&pp, pk_p.clone(), &label).unwrap())
+            },
+        );
+    }
+    proving_benchmarks.finish();
 
-        c.bench_with_input(
-            BenchmarkId::new(description.as_str(), degree),
+    let mut verifying_benchmarks = c.benchmark_group("verify");
+    for degree in MINIMUM_DEGREE..MAXIMUM_DEGREE {
+        let mut circuit = BenchCircuit::<_, EdwardsParameters>::new(degree);
+        let (pk_p, verifier_data) =
+            circuit.compile(&pp).expect("Unable to compile circuit.");
+        let proof = circuit.gen_proof(&pp, pk_p.clone(), &label).unwrap();
+        let VerifierData { key, pi_pos } = verifier_data;
+        verifying_benchmarks.bench_with_input(
+            BenchmarkId::from_parameter(degree),
             &degree,
-            |b, _degree| {
+            |b, _| {
                 b.iter(|| {
                     ark_plonk::circuit::verify_proof(
-                        black_box(&pp),
-                        black_box(verifier_data.clone().key()),
-                        black_box(&proof),
```

Comment:
Why removing the `black_box` here??

They prevent the compiler for optimizations which is exactly what we want inside of the benches.

Reply:
Black boxes are called around functions not function arguments. And the "benchmark with input" method that's used here automatically calls blackbox for you.